cli/cmd/agent_aws-install_ec2ssm.go

```diff
@@ -0,0 +1,267 @@
+
+To filter by instance tag:
+
+    lacework agent aws-install ec2ssm --tag TagName,TagValue
+
+To filter by instance tag key:
+
+    lacework agent aws-install ec2ssm --tag_key TagName
+
+To provide an agent access token of your choice, use the command 'lacework agent token list',
+select a token and pass it to the '--token' flag. This flag must be selected if the
+'--noninteractive' flag is set.
+
+    lacework agent aws-install ec2ssm --token <token>
+
+AWS credentials are read from the following environment variables:
+- AWS_ACCESS_KEY_ID
+- AWS_SECRET_ACCESS_KEY
+- AWS_SESSION_TOKEN (optional)
+- AWS_REGION (optional)`,
+	}
+)
+
+func init() {
+	// 'agent aws-install ec2ssm' flags
+	agentInstallAWSSSMCmd.Flags().StringVar(&agentCmdState.InstallTagKey,
+		"tag_key", "", "only install agents on infra with this tag key set",
+	)
+	agentInstallAWSSSMCmd.Flags().StringSliceVar(&agentCmdState.InstallTag,
+		"tag", []string{}, "only install agents on infra with this tag",
+	)
+	agentInstallAWSSSMCmd.Flags().StringSliceVarP(&agentCmdState.InstallIncludeRegions,
+		"include_regions", "r", []string{}, "list of regions to filter on",
+	)
+	agentInstallAWSSSMCmd.Flags().StringVar(&agentCmdState.InstallAgentToken,
+		"token", "", "agent access token",
+	)
+	agentInstallAWSSSMCmd.Flags().IntVarP(
+		&agentCmdState.InstallMaxParallelism,
+		"max_parallelism",
+		"n",
+		50,
+		"maximum number of workers executing AWS API calls, set if rate limits are lower or higher than normal",
+	)
+	agentInstallAWSSSMCmd.Flags().StringVar(
+		&agentCmdState.InstallBYORole,
+		"iam_role_name",
+		"",
+		"IAM role name (not ARN) with SSM policy, if not provided then an ephemeral role will be created",
+	)
+	agentInstallAWSSSMCmd.Flags().BoolVar(
+		&agentCmdState.InstallSkipCreatInfra,
+		"skip_infra_creation",
+		false,
+		"set this flag to skip IAM / SSM infra creation. Assumes all instances have SSM enabled",
+	)
+}
+
+func installAWSSSM(_ *cobra.Command, _ []string) error {
+	token := agentCmdState.InstallAgentToken
+	if token == "" {
+		if cli.InteractiveMode() {
+			// user didn't provide an agent token
+			cli.Log.Debugw("agent token not provided, asking user to select one now")
+			var err error
+			token, err = selectAgentAccessToken()
+			if err != nil {
+				return err
+			}
+		} else {
+			return errors.New("user did not provide or interactively select an agent token")
+		}
+	}
+
+	runners, err := awsDescribeInstances()
+	if err != nil {
+		return err
+	}
+
+	cfg, err := GetConfig()
+	if err != nil {
+		return err
+	}
+
+	var role types.Role
+	var instanceProfile types.InstanceProfile
+	if !agentCmdState.InstallSkipCreatInfra {
+		var err error
+		role, instanceProfile, err = SetupSSMAccess(cfg, agentCmdState.InstallBYORole, token)
+		defer func() {
+			err := TeardownSSMAccess(cfg, role, instanceProfile, agentCmdState.InstallBYORole) // clean up after ourselves
+			cli.Log.Warnw("got an error while tearing down IAM infra", "error", err)
```

Comment:
We should be more precise. `IAM infra` doesn't mean anything by itself.

```suggestion
			cli.Log.Warnw("got an error while tearing down IAM infra", "error", err)
```

Reply:
Actually, I think this is accurate (although I suppose we could change from `IAM infra` to `IAM role / instance profile`. This function does tear down the role and instance profile, if they were created. It does not detach the temporary instance profile. I think this is done automatically when the instance profile is deleted (https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_manage_delete.html).

Reply:
Sure, calling this `IAM role / instance profile` would be a good change.